### ISSUE TITLE
Alert js closing event name, doesn't

### ISSUE THREAD

**Issue by GeoSot** [twbs/bootstrap#33761]
I just noticed that `Alert.js` is the only of our components that emits an event named `closed.bs.alert` .
All the others use the `hidden.bs.{name}`

However the alert finally removes its self.

Are we ok with it or maybe should emit `disposed `or `hidden`?
The reality that it is disposing its self, and at least it could be streamlined with the rest.

I am just saying, that may it  would be the right moment to rename the event before stable 

PS1: after this, all disposals could emit an event,

**\CC** @twbs/js-review

**Comment by rohit2sharma95** [comment]
True but it's a breaking change, and currently there is no issue/request to emit the event for disposal of plugins? :slightly_smiling_face:

/CC @XhmikosR

**Comment by XhmikosR** [comment]
Yup, agreed it's a BC for sure. You should make an issue or add this in the v6.0.0 project so that we make the change then :)

**Comment by mdo** [comment]
This is an old ass issue, but—alerts are the only thing that are fully destroyed, so their event name _does_ make sense as we're not `show`-ing a hidden element, or hiding a `show`n element. But, I think we could update this to offer both full remove and a classic show/hide. Will take a pass.
